dipy/workflows/vis_registeration.py

```diff
@@ -0,0 +1,278 @@
+from __future__ import division, print_function, absolute_import
+from dipy.workflows.workflow import Workflow
+import logging
+import numpy as np
+import nibabel as nib
+from dipy.workflows.core import write_gif
+from dipy.io.image import load_affine_matrix
+from dipy.viz.regtools import overlay_slices
+from dipy.viz import (window, actor)
+
+from dipy.viz.window import snapshot
+
+
+class VisualizeRegisteredImage(Workflow):
+
+    def process_image_data(self, static_img, moved_img):
+
+        """
+        Function for pre-processing the image data. It involves
+        normalizing and copying static, moving images in the
+        red and green channel, respectively.
+
+        type_vis: str
+            The type of visualisation being created. This affects the
+            range of values to be selected from the moved image.
+
+        return: the normalized image data and range of pixels to be
+        selected.
+        """
+
+        static_img = 255 * ((static_img - static_img.min()) /
+                            (static_img.max() - static_img.min()))
+        moved_img = 255 * ((moved_img - moved_img.min()) /
+                           (moved_img.max() - moved_img.min()))
+
+        # Create the color images
+        overlay = np.zeros(shape=(static_img.shape) + (3,), dtype=np.uint8)
+        overlay[..., 0] = static_img
+        overlay[..., 1] = moved_img
+        mean, std = overlay[overlay > 0].mean(), overlay[overlay > 0].std()
+        value_range = (mean - 0.5 * std, mean + 1.5 * std)
+        return overlay, value_range
+
+    def get_row_cols(self, num_slices):
+
+        """
+        Experimetal helper function to get the number
+        of rows and columns for the mosaic.
+
+        num_slices: int
+            The number of slices as obtained from the
+            create_mosaic function.
+        return: the number of rows and columns.
+        """
+        rows, cols = 0, 0
+
+        while True:
+            if num_slices % 5 == 0:
+                break
+            num_slices += 1
+
+        for i in range(5, num_slices):
+
+            if num_slices % i == 0:
+                rows = i
+                cols = num_slices // i
+                break
+
+        return rows, cols
+
+    def create_mosaic(self, static_img, moved_img,
+                      affine, fname):
+        """
+        Function for creating the mosaic of the moved image. It
+        currently only supports slices from the axial plane.
+
+        fname: str, optional
+            Filename to be used for saving the mosaic (default 'mosaic.png').
+        """
+
+        overlay, value_range = self.process_image_data(static_img,
+                                                       moved_img)
+
+        renderer = window.Renderer()
+        renderer.background((0.5, 0.5, 0.5))
+
+        slice_actor = actor.slicer(overlay, affine, value_range)
+        renderer.projection('parallel')
+
+        cnt = 0
+        X, Y, num_slices = slice_actor.shape[:3]
```

Comment:
I thought PEP8 didn't like upper case names. Isn't there a warning?

Reply:
Yes, PEP8 don't like capital letter names but I kept it that way to remind me that these are the image dimensions. I was experimenting so there were small case x,y and z too. Reverted to the small case now.